src/compile.rs

```diff
@@ -171,7 +172,25 @@ cfg_if! {
             println!("\nDisassembly:");
             let insns = cs.disasm_all(&mem, 0x0).unwrap();
             for i in insns.iter() {
-                println!("{}", i);
+                let mut line = String::new();
+
+                write!(&mut line, "{:4x}:\t", i.address()).unwrap();
+
+                let mut bytes_str = String::new();
+                for b in i.bytes() {
+                    write!(&mut bytes_str, "{:02x} ", b).unwrap();
+                }
+                write!(&mut line, "{:21}\t", bytes_str).unwrap();
+
+                if let Some(s) = i.mnemonic() {
+                    write!(&mut line, "{}\t", s).unwrap();
+                }
+
+                if let Some(s) = i.op_str() {
+                    write!(&mut line, "{}", s).unwrap();
+                }
+
+                println!("{}", line);
```

Comment:
Neat! One additional thing that would be nice to do here is pass `print_disassembly` a subslice of the memory which is only `code_size` bytes long, so that we don't attempt to disassemble the jump table entries :-).

Reply:
Done and done. It also prints the readonly data afterward now. A simple `br_table` test file I have looks like this:

```
Disassembly of 76 bytes:
   0:	40 55                	push	rbp
   2:	48 89 e5             	mov	rbp, rsp
   5:	40 83 fe 0c          	cmp	esi, 0xc
   9:	40 73 10             	jae	0x1c
   c:	48 8d 05 39 00 00 00 	lea	rax, qword ptr [rip + 0x39]
  13:	48 63 0c b0          	movsxd	rcx, dword ptr [rax + rsi*4]
  17:	48 01 c8             	add	rax, rcx
  1a:	ff e0                	jmp	rax
  1c:	40 eb 24             	jmp	0x43
  1f:	40 b8 64 00 00 00    	mov	eax, 0x64
  25:	40 eb 21             	jmp	0x49
  28:	40 b8 65 00 00 00    	mov	eax, 0x65
  2e:	40 eb 18             	jmp	0x49
  31:	40 b8 66 00 00 00    	mov	eax, 0x66
  37:	40 eb 0f             	jmp	0x49
  3a:	40 b8 67 00 00 00    	mov	eax, 0x67
  40:	40 eb 06             	jmp	0x49
  43:	40 b8 68 00 00 00    	mov	eax, 0x68
  49:	40 5d                	pop	rbp
  4b:	c3                   	ret	

Followed by 48 bytes of read-only data:
   0:  ee ff ff ff  e5 ff ff ff  dc ff ff ff  d3 ff ff ff 
  16:  ee ff ff ff  e5 ff ff ff  dc ff ff ff  d3 ff ff ff 
  32:  ee ff ff ff  e5 ff ff ff  dc ff ff ff  d3 ff ff ff 
```